### ISSUE TITLE
Date picker label is not accessible / accessibility documentation lacking

### ISSUE THREAD

**Issue by benmccann** [mui-org/material-ui-pickers#1232]
## Description

Form labels must be accessible for blind users by including a `label` element with `for` attribute. Example:

```
<label for="firstname">First name:</label>
<input type="text" name="firstname" id="firstname">
```

When I run Chrome's accessibility audit, it tells me the calendar components are not accessible. It's possible that the calendar component is already accessible, but if so there's limited documentation. I did a google search `accessibility site:material-ui-pickers.dev`, which returned no results other than the homepage saying it was accessible. However, I couldn't find any other details.

## If you don't mind add a fun gif or meme, but no pressure 

![A GIF or MEME to give some spice of the internet](https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcSa-IHDQvDFkPBqdnL-wOEc2repoLHpVfxSaBKdF3VRKd4M0hY3)

## Environment

| Tech                 | Version |
| -------------------- | ------- |
| @material-ui/pickers | 3.2.1 |
| material-ui          | 4.1.1 |
| React                | 16.8.3 |
| Browser              | Chrome 75 |
| Peer library         | date-fns |  |

## Steps to reproduce

1. Add `KeyboardDatePicker` to page
2. Open Chrome Developer Tools. Switch to Audits tab. Run audit
3. Under Accessibility, an error is reported: "Form elements do not have associated labels"

## Expected behavior

I expect the inputs to be accessible. For a `TextField` all I need to do is ensure the element has a unique `id`

## Actual behavior

![Screenshot from 2019-08-05 09-23-41](https://user-images.githubusercontent.com/322311/62479646-cb7aa000-b762-11e9-80a6-5a9efc31aa03.png)

## Live example

I will work on deploying my application to production and will share when it's available

**Comment by dmtrKovalenko** [comment]
We are wrapping pure material-ui text field. So you will be able to pass any accessibility props the same as you will pass them to the material-ui text fiedl

**Comment by benmccann** [comment]
I've specified an `id` on my input as shown below. For `TextField` that's all I need to do. But it doesn't seem to work for `KeyboardDatePicker`
```
const StyledDobPicker = styled(KeyboardDatePicker)`
  label.Mui-focused {
    color: #e7396d;
  }
  .MuiOutlinedInput-root {
    &:hover fieldset {
      border-color: #e7396d;
    }
    &.focused fieldset {
      border-color: #e7396d;
    }
  }
  .MuiOutlinedInput-root.Mui-focused .MuiOutlinedInput-notchedOutline {
    border-color: #e7396d;
  }
  .MuiOutlinedInput-input {
    background-color: #ffffff;
  }
`;


                                <StyledDobPicker
                                    id="dob"
                                    label={this.props.lang().account.details.dob}
                                    format="dd/MM/yyyy"
                                    maxDate={subYears(new Date(), 18)}
                                    onChange={this.handleDob}
                                    classes={{ root: css.textField }}
                                    value={this.state.account.dob}
                                    inputVariant="outlined" />
```

**Comment by dmtrKovalenko** [comment]
Here is a screen with passed pure `id` prop into the picker. You can see that is it doing actually the same as [materila-ui TextField](https://material-ui.com/components/text-fields/#accessibility) doing
![image](https://user-images.githubusercontent.com/16926049/63213923-b4e92700-c11a-11e9-8a10-17b06652446b.png)
